pulp_container/app/registry.py

```diff
@@ -249,10 +242,9 @@ async def get_by_digest(self, request):
             raise PathNotResolved(path)
         else:
             artifact = ca.artifact
-            if artifact:
-                return await Registry._dispatch(artifact, headers)
-            else:
+            if not artifact and isinstance(ca_content, Blob):
                 return await self._stream_content_artifact(request, web.StreamResponse(), ca)
+            return await Registry._dispatch(artifact, headers)
```

Comment:
Do we need to consider any fallback in the case when there is no artifact present and yet the ca_content variable is not of the `Blob` type?

Reply:
We might end up returning HTTP 500 in such a case.

Reply:
How about returning an error like "the manifest could not be located, please, run the repair command"? I am not sure how container clients will cope with this.

Reply:
While there is an expectation that manifest would always have artifact downloaded even in the on_demand case, i forgot to remember our lovely 'reclaim space' feature that would potentially delete the artifact from the storage and db.

In that case we should probably focus then on how to get fixed the pulpcore downloader bug and close this PR.

Reply:
We could also add a safeguard check and raise 404 for when artifact is not found for type != blob ( that is to circumvent the downloader bug mostly..)

Reply:
I am all in for the safeguard check until we resolve the downloader issue.

Reply:
I will add path not resolved, but now I am not sure myself I like this, since behavior for blobs and manifests will differ.